pkg/diff/diff.go

```diff
@@ -322,3 +341,101 @@ func newMapPath(path []PathStep, index string) []PathStep {
 	})
 	return next
 }
+
+func (d *differ) addNode(path []PathStep, tx, ty reflect.Type, vx, vy reflect.Value) {
+	if len(d.ignoredPaths) > 0 {
+		if d.isIgnoredPaths(path) {
+			return
+		}
+
+		pathString := makePathString(path)
```

Comment:
How about removing `isIgnoredPathsForString` by calling `makePathStiring` or `path.String()` before that.
```suggestion
		pathString := makePathString(path)
		if d.isIgnoredPaths(pathString) {
			return
		}
```

Reply:
Your implementation is better. Thank you.